text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1067 @@
+
+#### Message `AttributeSet`
+
+AttributeSet represents a set of attributes. Multiple Samples, Locations and Mappings may have the same attributes and that's why this is a separate message. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `attributes`
+
+Attributes associated with a specific Sample, Location or a Mapping.
+attributes is a collection of key/value pairs. Note, global attributes
+like server name can be set using the resource API.
+
+##### Field `dropped_attributes_count`
+
+dropped_attributes_count is the number of attributes that were discarded. Attributes
+can be discarded because their keys are too long or because there are too many
+attributes. If this value is 0, then no attributes were dropped.
+</details>
+
+#### Message `Link`
+
+Represents a relationship between a Sample (ephemeral structure represented by references to a Stacktrace, AttributeSet, Link + value and a timestamp) and a trace span. This allows for linking between specific Samples within a profile and traces.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `trace_id`
+
+A unique identifier of a trace that this linked span is part of. The ID is a
+16-byte array.
+
+##### Field `span_id`
+
+A unique identifier for the linked span. The ID is an 8-byte array.
+</details>
+
+#### Message `Location`
+
+Describes function and line table debug information. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `mapping_index`
+
+The id of the corresponding profile.Mapping for this location.
+It can be unset if the mapping is unknown or not applicable for
+this profile type.
+
+##### Field `address`
+
+The instruction address for this location, if available.  It
+should be within [Mapping.memory_start...Mapping.memory_limit]
+for the corresponding mapping. A non-leaf address may be in the
+middle of a call instruction. It is up to display tools to find
+the beginning of the instruction if necessary.
+
+##### Field `line`
+
+Multiple line indicates this location has inlined functions,
+where the last entry represents the caller into which the
+preceding entries were inlined.
+E.g., if memcpy() is inlined into printf:
+line[0].function_name == "memcpy"
+line[1].function_name == "printf"
+
+##### Field `attribute_set_index`
+
+Reference to an attribute set from the Profile's attribute set table.
+</details>
+
+#### Message `Mapping`
+
+Describes the mapping from a binary to its original source code. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+
```

Comment:
Operating systems, like Windows, allow mappings to come and go. Typically, we collect mappings with a "load/unload" event with a timestamp. It seems like in this format while writing the data the linking of the load/unload times vs the sample times to determine the right mapping for a given process must be done ahead of time.